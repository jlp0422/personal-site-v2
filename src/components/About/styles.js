import styled from '@emotion/styled'

export const Copy = styled.p`
  font-size: 2rem;
  padding: 0.5rem 0;
`

export const ContactList = styled.div`
  > div {
    font-family: ${({ theme }) => theme.fonts.primary};
    font-size: 1.8rem;
  }
  span,
  a {
    font-size: 1.8rem;
  }
  a {
    cursor: pointer;
  }
`

export const ContentWrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr 350px;
  grid-column-gap: 3%;
  @media only screen and (max-width: 800px) {
    grid-template-columns: 1fr 250px;
  }
  @media only screen and (max-width: 640px) {
    display: block;
  }
`

export const ImageWrapper = styled.div`
  @media only screen and (max-width: 640px) {
    width: 300px;
    margin: 2rem auto;
  }
`
