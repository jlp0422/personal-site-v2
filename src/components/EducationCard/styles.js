import styled from '@emotion/styled'
import {
  CardFlipBack,
  CardFlipContainer,
  CardFlipInner,
  CardShared,
  ExpInfo,
  ExpTitle
} from '../shared'

export const Container = styled(CardFlipContainer)`
  text-align: center;
  @media only screen and (max-width: 640px) {
    height: 250px;
  }
`

export const InnerContainer = styled(CardFlipInner)`
  box-shadow: ${({ theme }) => theme.palette.boxShadow.dark};
`

export const CardFront = styled(CardShared)`
  background-color: ${({ theme }) => theme.palette.black.darkest};
`

export const CardBack = styled(CardFlipBack)`
  background-color: ${({ theme }) => theme.palette.white.lighter};
`

export const Title = ExpTitle

export const Info = styled(ExpInfo)`
  width: 90%;
  grid-template-columns: repeat(2, 1fr);
`
