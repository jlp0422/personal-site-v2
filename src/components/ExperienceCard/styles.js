import styled from '@emotion/styled'
import {
  ExpInfo,
  ExpTitle,
  CardFlipContainer,
  CardFlipInner,
  CardFlipBack,
  CardShared
} from '../shared'

export const Container = styled(CardFlipContainer)`
  display: grid;
  text-align: center;
  margin: 2.5rem auto;
  @media only screen and (max-width: 640px) {
    height: 250px;
  }
`

export const InnerContainer = styled(CardFlipInner)`
  box-shadow: ${({ theme }) => theme.palette.boxShadow.light}
`

export const CardFront = styled(CardShared)`
  padding: 1.5rem;
  background-color: ${({ theme }) => theme.palette.black.darkest};
`

export const CardBack = styled(CardFlipBack)`
  padding: 1.5rem;
  background-color: ${({ theme }) => theme.palette.white.lighter}
`

export const Title = ExpTitle

export const Info = styled(ExpInfo)`
  width: 100%;
  align-items: center;
  @media only screen and (max-width: 640px) {
    grid-template-columns: 1fr;
  }
`

export const LogoContainer = styled.div`
  display: block;
  visibility: visible;
  width: 60px;
  margin: 1rem auto 0;
  @media only screen and (max-width: 640px) {
    display: none;
    visibility: hidden;
  }
`
