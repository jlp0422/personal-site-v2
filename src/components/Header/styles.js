import styled from '@emotion/styled'
import { Link } from 'gatsby'
import { ActiveLink } from '../../components/shared/Styled'

export const StyledHeader = styled.header`
  background: ${({ theme }) => theme.palette.link.secondary};
  margin-bottom: 1.5rem;
`

export const Div = styled.div`
  margin: 0 auto;
  max-width: 1200px;
  padding: 2rem 1.2rem;
  display: grid;
  grid-template-columns: 120px 120px 1fr 120px 120px;
  justify-items: center;
  align-items: center;
  grid-column-gap: 2rem;
  @media only screen and (max-width: 640px) {
    padding: 0.8rem;
    display: flex;
    align-items: center;
    justify-content: center;
  }
`

export const H1 = styled.h1`
  margin: 0;
  text-align: center;
  text-transform: uppercase;
`

export const HeaderLink = styled(Link)`
  letter-spacing: 0.3rem;
  font-family: ${({ theme }) => theme.fonts.header};
  color: white;
  text-decoration: none;
  font-size: 4.5rem;
  @media only screen and (max-width: 800px) {
    visibility: hidden;
    display: none;
  }
`

export const LogoLink = styled(Link)`
  margin: 0;
  visibility: hidden;
  display: none;
  @media only screen and (min-width: 640px) and (max-width: 800px) {
    visibility: visible;
    display: block;
  }
`

export const NavLink = styled(ActiveLink)`
  @media only screen and (max-width: 640px) {
    display: none;
    visibility: hidden;
  }
`

export const MobileLogo = styled.button`
  margin: 0;
  visibility: hidden;
  display: none;
  @media only screen and (max-width: 639px) {
    visibility: visible;
    display: block;
  }
`

export const List = styled.ul`
  padding: 0 0 1rem;
  list-style: none;
  margin-top: 0;
  visibility: hidden;
  display: none;
  @media only screen and (max-width: 640px) {
    visibility: visible;
    display: block;
  }
`
