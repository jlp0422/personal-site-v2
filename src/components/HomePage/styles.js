import { keyframes } from '@emotion/core'
import styled from '@emotion/styled'
import { Link } from 'gatsby'

export const spin = keyframes`
  from {
    transform: rotate(0deg)
  }

  to {
    transform: rotate(360deg)
  }
`

export const GridContainer = styled.div`
  display: grid;
  grid-template: repeat(2, 1fr) / repeat(2, 1fr);
  grid-gap: 2rem 2rem;
  justify-content: center;
  align-items: center;
  text-align: center;
  margin-top: 1rem;
  > * {
    margin: 0;
  }
  @media only screen and (max-width: 800px) {
    grid-template: repeat(4, 1fr) / 1fr;
  }
`

export const LinkBox = styled(Link)`
  background-color: ${({ color }) => color};
  padding: 8rem;
  font-family: ${({ theme }) => theme.fonts.nav};
  text-decoration: none;
  transition: all 0.2s linear;
  > h3 {
    margin: 1.5rem;
  }
  @media only screen and (min-width: 641px) {
    :hover {
      transform: scale(1.05);
      div {
        animation: ${spin} 2s linear infinite;
      }
    }
  }
  @media only screen and (max-width: 800px) {
    padding: 5rem;
  }
  @media only screen and (max-width: 640px) {
    padding: 1rem;
    > h3 {
      margin: 0.5rem;
    }
  }
`
