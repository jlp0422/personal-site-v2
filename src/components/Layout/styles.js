import styled from '@emotion/styled'

export const ContainerDiv = styled.div`
  margin: 0 auto;
  max-width: 1200px;
  padding: 0 1.5rem 1.5rem;
  min-height: 85vh;
  display: grid;
  grid-template-rows: 1fr auto;
`
