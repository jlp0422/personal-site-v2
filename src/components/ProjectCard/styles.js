import styled from '@emotion/styled'

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background-color: white;
  margin: 1rem 0;
  padding: 1.5rem;
  border: 1px solid black;
  transition: all 0.2s linear;
  --webkit-transition: all 0.2s linear;
  :hover {
    box-shadow: ${({ theme }) => theme.palette.boxShadow.primary};
    transform: translate(-8px, -8px);
    --webkit-transform: translate(-8px, -8px);
  }
  @media only screen and (max-width: 480px) {
    box-shadow: ${({ theme }) => theme.palette.boxShadow.primary};
    :hover {
      transform: none;
    }
  }
`

export const LinkContainer = styled.div`
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  * + * {
    margin-top: 0;
  }
`
