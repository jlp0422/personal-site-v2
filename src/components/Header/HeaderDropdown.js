import { css } from '@emotion/core'
import React from 'react'
import { ActiveLink } from '../../components/shared'
import { List } from './styles'
import ResumeLink from '../Resume/ResumeLink'
import { Fragment } from 'react'

const mobileNavLinks = [
  { label: 'About', path: '/about' },
  { label: 'Experience', path: '/experience' },
  { label: 'Projects', path: '/projects' },
  {
    label: 'Resume',
    path: '/resume',
    Component: ResumeLink,
    element: 'span'
  }
]

const Dropdown = ({ toggleDropdown, isDropdownOpen }) => {
  const toggle = () => toggleDropdown(!isDropdownOpen)
  if (!isDropdownOpen) {
    return null
  }
  return (
    <List>
      {mobileNavLinks.map(({ label, path, Component, ...rest }) => {
        const LinkRenderer = Component || Fragment
        return (
          <li
            css={css`
              text-align: center;
              border-bottom: none;
            `}
            key={path}
          >
            <LinkRenderer>
              <ActiveLink
                onKeyDown={toggle}
                onClick={toggle}
                to={path}
                {...rest}
              >
                {label}
              </ActiveLink>
            </LinkRenderer>
          </li>
        )
      })}
    </List>
  )
}

export default Dropdown
