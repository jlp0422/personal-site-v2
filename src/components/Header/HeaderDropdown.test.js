import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeWrapper } from '../shared'
import HeaderDropdown from './HeaderDropdown'
import { NavLink, List } from './styles'

describe('HeaderDropdown', () => {
  const getOutput = (isDropdownOpen = true) =>
    renderer.create(
      <ThemeWrapper>
        <HeaderDropdown isDropdownOpen={isDropdownOpen} />
      </ThemeWrapper>
    )

  it('Renders correctly when the isDropdownOpen prop is true', () => {
    const tree = getOutput().toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders nothing when the isDropdownOpen prop is false', () => {
    const instance = getOutput(false).root
    const list = instance.findAllByType(List)
    expect(list).toHaveLength(0)
  })

  it('Displays the NavLinks in the correct order', () => {
    const instance = getOutput().root
    const navLinks = instance.findAllByType('li')
    expect(navLinks).toHaveLength(4)

    const getDeepProps = link => link.props.children.props.children.props
    const getLinkAt = index => getDeepProps(navLinks[index])

    navLinks.forEach(link => {
      expect(getDeepProps(link)).toHaveProperty('to')
      expect(getDeepProps(link)).toHaveProperty('children')
      expect(getDeepProps(link)).toHaveProperty('onKeyDown')
      expect(getDeepProps(link)).toHaveProperty('onClick')
    })
    expect(getLinkAt(0)).toEqual(
      expect.objectContaining({ to: '/about', children: 'About' })
    )
    expect(getLinkAt(1)).toEqual(
      expect.objectContaining({ to: '/experience', children: 'Experience' })
    )
    expect(getLinkAt(2)).toEqual(
      expect.objectContaining({ to: '/projects', children: 'Projects' })
    )
    expect(getLinkAt(3)).toEqual(
      expect.objectContaining({
        to: '/resume',
        children: 'Resume',
        element: 'span'
      })
    )
  })
})
