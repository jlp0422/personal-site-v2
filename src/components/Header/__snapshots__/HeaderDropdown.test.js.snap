// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderDropdown Renders correctly when the isDropdownOpen prop is true 1`] = `
<ul
  className="css-1k3rwkt"
>
  <li
    css={
      Object {
        "map": undefined,
        "name": "1pwn70r",
        "next": undefined,
        "styles": "
              text-align: center;
              border-bottom: none;
            ",
        "toString": [Function],
      }
    }
  >
    <a
      className="css-m9trjp"
      href="/about"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      About
    </a>
  </li>
  <li
    css={
      Object {
        "map": undefined,
        "name": "1pwn70r",
        "next": undefined,
        "styles": "
              text-align: center;
              border-bottom: none;
            ",
        "toString": [Function],
      }
    }
  >
    <a
      className="css-m9trjp"
      href="/experience"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      Experience
    </a>
  </li>
  <li
    css={
      Object {
        "map": undefined,
        "name": "1pwn70r",
        "next": undefined,
        "styles": "
              text-align: center;
              border-bottom: none;
            ",
        "toString": [Function],
      }
    }
  >
    <a
      className="css-m9trjp"
      href="/projects"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      Projects
    </a>
  </li>
  <li
    css={
      Object {
        "map": undefined,
        "name": "1pwn70r",
        "next": undefined,
        "styles": "
              text-align: center;
              border-bottom: none;
            ",
        "toString": [Function],
      }
    }
  >
    <a
      className="css-3wo2qn"
      href="/static/f1ac7cc7de0bcb8ba3e02cc0d5662ddc/Philipson-Jeremy-Resume.pdf"
      target="_blank"
    >
      <span
        className="css-m9trjp"
        onClick={[Function]}
        onKeyDown={[Function]}
        to="/resume"
      >
        Resume
      </span>
    </a>
  </li>
</ul>
`;
